4/Dockerfile

```diff
@@ -1,35 +1,31 @@
-FROM debian:stretch-slim
+ARG NODEJS_VERSION=10.18.0
+FROM node:${NODEJS_VERSION}-stretch-slim
 
+ARG SONAR_SCANNER_VERSION=4.2.0.1873
 ARG SONAR_SCANNER_HOME=/opt/sonar-scanner
-ARG NODEJS_HOME=/opt/nodejs
-ARG UID=1000
-ARG GID=1000
+ARG TYPESCRIPT_VERSION=3.7.4
+ARG UID=1001
+ARG GID=1001
```

Comment:
In our Jenkins env we rely on the UID 1000 (same as `jenkins/jenkins:lts`), otherwise sonar scanner will create undeniable file (as UID would be unknown).